File: ABB-setup/dv-processing-rel_1.7/docs/source/assets/code_samples/c++/reading_data/sample3.cpp
#include <dv-processing/io/camera_capture.hpp>

#include <chrono>

int main() {
	using namespace std::chrono_literals;

	// Open any camera
	dv::io::CameraCapture capture;

	// Run the loop while camera is still connected
	while (capture.isRunning()) {
		// Read batch of events, check whether received data is correct.
		// The method does not wait for data arrive, it returns immediately with
		// the latest available data or if no data is available, returns a `std::nullopt`.
		if (const auto events = capture.getNextEventBatch(); events.has_value()) {
			// Print received packet information
			std::cout << *events << std::endl;
		}
		else {
			// No data has arrived yet, short sleep to reduce CPU load.
			std::this_thread::sleep_for(1ms);
		}
	}

	return 0;
}

File: ABB-setup/dv-processing-rel_1.7/docs/source/assets/code_samples/python/reading_data/sample3.py
import dv_processing as dv
import time

# Open any camera
capture = dv.io.CameraCapture()

# Run the loop while camera is still connected
while capture.isRunning():
    # Read batch of events
    events = capture.getNextEventBatch()

    # The method does not wait for data arrive, it returns immediately with
    # latest available data or if no data is available, returns a `None`
    if events is not None:
        # Print received packet time range
        print(f"Received events within time range [{events.getLowestTime()}; {events.getHighestTime()}]")
    else:
        # No data has arrived yet, short sleep to reduce CPU load
        time.sleep(0.001)
